A84-xds-authority-literal-rewrite.md

```diff
@@ -0,0 +1,72 @@
+Title
+----
+* Author(s): Zhiyan Foo
+* Approver: a11r
+* Status: Draft
+* Implemented in: TBD
+* Last updated: 2024-11-06
+* Discussion at: TBD (filled after thread exists)
+
+## Abstract
+
+Propose extending gRPC's xDS client to honour `host_rewrite_literal` in route configuration for
+authority header rewriting when `trusted_xds_server` is set.
+
+## Background
+
+gRPC xDS currently supports `auto_host_rewrite` as per gRFC A81 but lacks `host_rewrite_literal`
+support, useful when a single cluster targets a reverse proxy routing based on the authority
+header.
+```
+            +------------------+
+            |  gRPC xDS Client |  <- (RDS configuration sets `host_rewrite_literal` to
+            +------------------+      target one of Svc A/B/C)
+                     |
+                     |    <- (gRPC xDS client reuses the same CDS target for all connections to
+                     |        the proxy)
+                 +---------+
+                 |  Proxy  |
+                 +---------+
+                      |   <- (Proxy routing logic based on Authority header)
+      .---------------+--------------.
+     /                |               \
++--------+       +--------+       +--------+
+| Svc A  |       | Svc B  |       | Svc C  |
++--------+       +--------+       +--------+
+```
+
+### Related Proposals:
+* [A81: `auto_host_rewrite` in gRPC xDS](https://github.com/grpc/proposal/blob/master/A81-xds-routes.md)
+* [A86: xDS-Based HTTP CONNECT](https://github.com/grpc/proposal/pull/455)
+
+## Proposal
+
+Implement `host_rewrite_literal` feature in gRPC xDS client, enabling explicit authority header
+rewrites. This should be conditioned on `trusted_xds_server`.
+
```

Comment:
You'll need to explain exactly which component of gRPC xDS are affected by this change: is setting the authority done in the config selector? Where is the authority currently set? Is there a chance that there are bad interops between the other places where we write an authority header? How will this interact with other settings such as `auto_host_rewrite`? We need to validate the config to make sure both are not set I think.

I would map those paragraphs the same way A81 does:
### xDS Resource Validation
### xDS ConfigSelector Changes
### xds_cluster_impl LB Policy Changes